core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -1672,7 +1675,25 @@ else if (node.getScope() == Scope.LOCAL) {
                                 "hashColumn", formatHash(node.getPartitioningScheme().getHashColumn())),
                         context);
             }
-            return processChildren(node, new Context(context.types()));
+            return processChildren(node, new Context(context.types(), context.isInitialPlan()));
+        }
+
+        @Override
+        public Void visitAdaptivePlanNode(AdaptivePlanNode node, Context context)
```

Comment:
I think we agreed that we print Initial and Current plan as two separate trees? 
Not render AdaptivePlanNodes. 

Was that not readable?

Reply:
Please file a ticket and add a todo in the code linking to the ticke